### ISSUE TITLE
Enhanced Referral Information

### ISSUE THREAD

**Issue by btsfav** [oxarbitrage/bitshares-explorer#6]
per account, this need to be cached somewhere I think.

I'd like to see:

* Total Referred
* Total LTM Registrations
* Total Earnings

**Comment by kimchi-king** [comment]
Yes, yes, yes. This type of data is important for those who actively promote the BTS system via their referral link. Otherwise, it's damn near impossible to validate your efforts.

**Comment by oxarbitrage** [comment]
i understand the importance of this and i was already looking a bit into it. if you check an account page like http://bitshares-explorer.io:9009/#/accounts/alfredo-worker you will see at the bottom there is a slot for referrer info. i was planning on adding each account the user has subscribed there.

we can just add totals as fav suggested but eventually i am pretty sure we will need the precise data. the problem is that i don't see other way to do it than looping throw all the accounts, checking the referrer they have and save the data in an offchain database. this is an intense computation that haves to be made and as other parts of the explorer it will initially be updated once a day only.

i plan to add this this week and update here. suggestions welcome.

**Comment by oxarbitrage** [comment]
with all the limitations we have core side in order to get good referrer data i was able to at least add the referrer count and account names, check at the bottom of:

http://bitshares-explorer.io:9009/#/accounts/kimchi-king

and

http://bitshares-explorer.io:9009/#/accounts/fav

this info is updated every 24 hours. in order to get total earnings we are going to need to wait for the elasticsearch plugin to be released(https://github.com/bitshares/bitshares-core/pull/444) because we need to go inside every operation done by the referrer, get paid fees and get the % of that fee that was paid to the referral account.

i hope it helps at least a bit.

**Comment by oxarbitrage** [comment]
this is also related https://github.com/bitshares/bitshares-core/issues/452

**Comment by btsfav** [comment]
https://github.com/bitshares/bitshares-ui/issues/661#issuecomment-342448275 maybe relevant

**Comment by btsfav** [comment]
@oxarbitrage maybe you could write the feature request for core?
